Naming/src/main/java/com/epam/engx/cleancode/naming/task5/FileRetriever.java

```diff
@@ -0,0 +1,87 @@
+package com.epam.engx.cleancode.naming.task5;
+
+import com.epam.engx.cleancode.naming.task5.thirdpartyjar.InvalidDirectoryException;
+import com.epam.engx.cleancode.naming.task5.thirdpartyjar.InvalidFileTypeException;
+import com.epam.engx.cleancode.naming.task5.thirdpartyjar.PropertyUtil;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+public final class FileRetriever {
+
+    private static final String[] IMAGES_EXTENSIONS = {"jpg", "png"};
+    private static final String[] DOCUMENTS_EXTENSIONS = {"pdf", "doc"};
+    private String basePath = PropertyUtil.loadProperty("basePath");
+
+    public File retrieveFile(String fileName) {
+        validateFileType(fileName);
+        final String directoryPath = basePath + File.separator;
+        return Paths.get(directoryPath, fileName).toFile();
+    }
+
+    private void validateFileType(String fileName) {
+        if (isInvalidFileType(fileName)) {
+            throw new InvalidFileTypeException("File type not Supported: " + fileName);
+        }
+    }
+
+    private boolean isInvalidFileType(String fileName) {
+        return isInvalidImage(fileName) && isInvalidDocument(fileName);
+    }
+
+    private boolean isInvalidImage(String fileName) {
+        FileExtensionChecker imageExtensions = new FileExtensionChecker(IMAGES_EXTENSIONS);
+        return !imageExtensions.test(fileName);
+    }
+
+    private boolean isInvalidDocument(String fileName) {
+        FileExtensionChecker documentExtensions = new FileExtensionChecker(DOCUMENTS_EXTENSIONS);
+        return !documentExtensions.test(fileName);
+    }
+
+    public List<String> retrieveAllImages() {
+        return retrieveAllowedFiles(basePath, IMAGES_EXTENSIONS);
+    }
+
+    public List<String> retrieveAllDocuments() {
+        return retrieveAllowedFiles(basePath, DOCUMENTS_EXTENSIONS);
+    }
+
+    private List<String> retrieveAllowedFiles(String directoryPath,
+                                              String[] allowedExtensions) {
+        final FileExtensionChecker fileExtensionChecker =
+                new FileExtensionChecker(allowedExtensions);
+        return Arrays.asList(retrieveFileFromDirectory(directoryPath)
+                .list(retrieveFilenameFilterByFileExtensionChecker(fileExtensionChecker)));
+    }
+
+    private File retrieveFileFromDirectory(String directoryPath) {
```

Comment:
```suggestion
    private File retrieveFile(String directoryPath) {
```

Reply:
The method has private access. At the same time, the class contains the public method "retrieveFile"